libsolidity/formal/SMTChecker.cpp

```diff
@@ -605,12 +606,12 @@ void SMTChecker::checkBooleanNotConstant(Expression const& _condition, string co
 		return;
 
 	m_interface->push();
-	m_interface->addAssertion(expr(_condition));
+	m_interface->addAssertion(currentPathConditions() && expr(_condition));
```

Comment:
Can you add a helper for `m_interface->addAssertion(currentPathCondition() && x)`?

Reply:
Done.